buildSrc/src/main/java/com/osacky/flank/gradle/YamlWriter.kt

```diff
@@ -27,167 +29,92 @@ internal class YamlWriter {
       if (base.instrumentationApk.isPresent) {
         appendln("  test: ${base.instrumentationApk.get()}")
       }
-      appendln(deviceString)
+      if (config.devices.isPresentAndNotEmpty) appendln(createDeviceString(config.devices.get()))
```

Comment:
Is the device a required property in flank? What would happen if it were omitted?

I believe the previous logic would alway add it, this logic only add it if it is present. Why the change?

Reply:
Actually not, we can omit devices and `NexusLowRes` will be taken as default. TBH the only required properties (from flank's perspective) is app apk and test apk. I have in my mind to change it in fladle but in separate PR

Reply:
But I can change implementation to reflect the previous one, and change it with another PR

Reply:
Makes sense. What version of the API/Android SDK does Flank use? I want to ensure that there are no unexpected changes introduced with this change.

Reply:
> Makes sense. What version of the API/Android SDK does Flank use? I want to ensure that there are no unexpected changes introduced with this change.

28

https://github.com/Flank/flank/blob/master/test_runner/src/main/kotlin/ftl/config/FtlConstants.kt#L47-L48